zeppelin-server/src/main/java/org/apache/zeppelin/utils/SecurityUtils.java

```diff
@@ -179,12 +274,210 @@ public static Collection getRealmsList() {
+    List<String> userList = new ArrayList<>();
+    String userDnTemplate = r.getUserDnTemplate();
+    String userDn[] = userDnTemplate.split(",", 2);
+    String userDnPrefix = userDn[0].split("=")[0];
+    String userDnSuffix = userDn[1];
+    JndiLdapContextFactory cf = (JndiLdapContextFactory) r.getContextFactory();
+    try {
+      LdapContext ctx = cf.getSystemLdapContext();
+      SearchControls constraints = new SearchControls();
+      constraints.setCountLimit(numUsersToFetch);
+      constraints.setSearchScope(SearchControls.SUBTREE_SCOPE);
+      String[] attrIDs = {userDnPrefix};
+      constraints.setReturningAttributes(attrIDs);
+      NamingEnumeration result = ctx.search(userDnSuffix, "(" + userDnPrefix + "=*" + searchText +
+          "*)", constraints);
+      while (result.hasMore()) {
+        Attributes attrs = ((SearchResult) result.next()).getAttributes();
+        if (attrs.get(userDnPrefix) != null) {
+          String currentUser = attrs.get(userDnPrefix).toString();
+          userList.add(currentUser.split(":")[1].trim());
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.error("Error retrieving User list from Ldap Realm", e);
+    }
+    LOGGER.info("UserList: " + userList);
+    return userList;
+  }
+
+  /**
+   * Function to extract users from Zeppelin LdapRealm.
+   */
+  private static List<String> getUserList(LdapRealm r, String searchText, int numUsersToFetch) {
+    List<String> userList = new ArrayList<>();
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("SearchText: " + searchText);
+    }
+    String userAttribute = r.getUserSearchAttributeName();
+    String userSearchRealm = r.getUserSearchBase();
+    String userObjectClass = r.getUserObjectClass();
+    JndiLdapContextFactory cf = (JndiLdapContextFactory) r.getContextFactory();
+    try {
+      LdapContext ctx = cf.getSystemLdapContext();
+      SearchControls constraints = new SearchControls();
+      constraints.setSearchScope(SearchControls.SUBTREE_SCOPE);
+      constraints.setCountLimit(numUsersToFetch);
+      String[] attrIDs = {userAttribute};
+      constraints.setReturningAttributes(attrIDs);
+      NamingEnumeration result = ctx.search(userSearchRealm, "(&(objectclass=" +
+          userObjectClass + ")("
+          + userAttribute + "=*" + searchText + "*))", constraints);
+      while (result.hasMore()) {
+        Attributes attrs = ((SearchResult) result.next()).getAttributes();
+        if (attrs.get(userAttribute) != null) {
+          String currentUser;
+          if (r.getUserLowerCase()) {
+            LOGGER.debug("userLowerCase true");
+            currentUser = ((String) attrs.get(userAttribute).get()).toLowerCase();
+          } else {
+            LOGGER.debug("userLowerCase false");
+            currentUser = (String) attrs.get(userAttribute).get();
+          }
+          if (LOGGER.isDebugEnabled()) {
```

Comment:
btw, why it is calling `isDebugEnabled`?

Reply:
I only did refactoring in this PR, didn't change it. It is as before, but I can improve it.